src/Pdk/Hooks/PdkProductSettingsHooks.php

```diff
@@ -92,23 +123,25 @@ public function savePdkProduct($post, int $productId): void
     {
         $appInfo = Pdk::getAppInfo();
 
-        $productSettingKeys = Arr::where($post, static function ($_, string $key) use ($appInfo) {
-            return Str::startsWith($key, "$appInfo->name-");
+        $productSettingKeys = Arr::where($post, static function ($_, string $key) use ($appInfo, $productId) {
+            return Str::startsWith($key, "$appInfo->name-childProductSettings--$productId-");
         });
 
+        if (empty($productSettingKeys)) {
```

Comment:
Dit is wel een beetje lomp nu er een extra laag is, kan je dit wat eleganter maken? bijv:

```php
$productSettingKeys = $this->getChildProductSettingsKeys($post, productId) ?? $this->getProductSettingsKeys($post, $productId);
```

En dan in de implementatie wat moois bedenken op die gedupliceerde `Arr::where` en `Str::startsWith` slierten